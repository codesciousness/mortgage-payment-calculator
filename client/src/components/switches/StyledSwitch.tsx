import React from 'react';
import AndroidSwitch from './AndroidSwitch';

type SwitchProps = {
    name: string;
    checked: boolean;
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

const StyledSwitch = ({ name, checked, onChange }:  SwitchProps): JSX.Element => {
    const id = name.replaceAll(' ', '');

    return (
        <AndroidSwitch
            id={id}
            checked={checked}
            onChange={onChange}
            sx={{
                '& .MuiSwitch-switchBase.Mui-checked + .MuiSwitch-track': {
                    background: 'linear-gradient(to left, SkyBlue, DeepSkyBlue)',
                },
                '& .MuiSwitch-thumb': {
                    backgroundColor: 'DeepSkyBlue',
                },
                '& .MuiSwitch-track': {
                    backgroundColor: 'LightSkyBlue',
                },
            }}
        />
    );
};

export default StyledSwitch;