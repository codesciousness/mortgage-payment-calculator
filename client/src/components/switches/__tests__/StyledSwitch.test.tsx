import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import StyledSwitch from '../StyledSwitch';

const name = 'Label Name';
const checkedTrue = true;
const checkedFalse = false;
const onChange = () => null;

describe('<StyledSwitch />', () => {
    test('StyledSwitch component is defined', () => {
        expect(<StyledSwitch name={name} checked={checkedFalse} onChange={onChange}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<StyledSwitch name={name} checked={checkedTrue} onChange={onChange}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly when checked prop is true', () => {
        render(<StyledSwitch name={name} checked={checkedTrue} onChange={onChange}/>);
        expect(screen.getByRole('checkbox')).toBeChecked();
    });
    test('renders correctly when checked prop is false', () => {
        render(<StyledSwitch name={name} checked={checkedFalse} onChange={onChange}/>);
        expect(screen.getByRole('checkbox')).not.toBeChecked();
    });
});