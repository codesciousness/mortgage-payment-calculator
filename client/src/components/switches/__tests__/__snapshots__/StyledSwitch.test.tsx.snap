// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledSwitch /> component matches the snapshot 1`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-1xp0olk-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      id="LabelName"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;
